.github/workflows/build-and-test.yaml

```diff
@@ -2,8 +2,11 @@ name: Build-and-test
 
 on:
   push:
+    branches:
```

Comment:
Consider adding a filter for tags in addition to branches if you plan to use tags for releases or other significant events. This can be done by adding a 'tags' section under 'push' and 'pull_request'. [medium]